app/schema/schema.go

```diff
@@ -139,15 +144,118 @@ func Action_GoCodegen(args *cli.Context) error {
 	outputDir := args.Path("output")
 	pkgName := args.String("package")
 
-	if generator != "go-gengo" {
+	switch generator {
+	case "go-gengo":
+		if err := os.MkdirAll(outputDir, os.ModePerm); err != nil {
+			return err
+		}
+		a := gengo.AdjunctCfg{}
+		gengo.Generate(outputDir, pkgName, ts, &a)
+	case "go-bindnode":
+		if err := generateGoBindnode(schemaFilePath, outputDir, pkgName, &ts); err != nil {
+			return err
+		}
+	default:
 		return fmt.Errorf("unsupported generator: %s", generator)
 	}
 
+	return nil
+}
+
+func generateGoBindnode(schemaFilePath, outputDir, pkgName string, ts *schema.TypeSystem) error {
 	if err := os.MkdirAll(outputDir, os.ModePerm); err != nil {
 		return err
 	}
-	a := gengo.AdjunctCfg{}
-	gengo.Generate(outputDir, pkgName, ts, &a)
 
-	return nil
+	// generate the basic Go types in types.go
+	f, err := os.Create(filepath.Join(outputDir, "types.go"))
+	if err != nil {
+		return err
+	}
+
+	if _, err := fmt.Fprintf(f, "package %s\n\n", pkgName); err != nil {
+		return err
+	}
+
+	delete(ts.GetTypes(), "Link")
+	ts.GetTypes()["Bogus"] = schema.SpawnBool("Bogus")
+	copy(ts.Names()[5:], ts.Names()[6:])
+	ts.Names()[len(ts.Names())-1] = "Bogus"
+
+	if err := bindnode.ProduceGoTypes(f, ts); err != nil {
+		return err
+	}
+
+	if err := f.Close(); err != nil {
+		return err
+	}
+
+	// generate schema prototypes in schema.go
+	relPath, err := filepath.Rel(outputDir, schemaFilePath)
+	if err != nil {
+		// TODO: better err
+		return err
+	}
+
+	type tmplfillIn struct {
+		PkgName         string
+		SchemaEmbedPath string
+		TypeNames       []string
+	}
+
+	tmpl, err := template.New("schematypegen").Parse(`
+package {{.PkgName}}
+
+import (
+	_ "embed"
+	
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/node/bindnode"
+	"github.com/ipld/go-ipld-prime/schema"
+)
+
+//go:embed {{.SchemaEmbedPath}}
+var embeddedSchema []byte
+
+var Types schemaSlab
+
+type schemaSlab struct {
+{{range .TypeNames}}
+{{.}}	schema.TypedPrototype{{end}}
+}
+
+func init() {
+	ts, err := ipld.LoadSchemaBytes(embeddedSchema)
+	if err != nil {
+		panic(err)
+	}
+{{range .TypeNames}}
+
+	Types.{{.}} = bindnode.Prototype(
+		(*{{.}})(nil),
+		ts.TypeByName("{{.}}"),
+	){{end}}
```

Comment:
Possibly this bit about having a struct full of convenient access to prototypes should be optional?  Bindnode can be used without it.  (Gengo has a table like this, but _can't_ be used without it.)

Or if we do have this table, perhaps it should be made by the upstream bindnode gen code?